test/addons-napi/test_typedarray/test_typedarray.c

```diff
@@ -136,6 +136,25 @@ napi_value CreateTypedArray(napi_env env, napi_callback_info info) {
   NAPI_CALL(env, napi_get_typedarray_info(
       env, input_array, &type, &length, NULL, &in_array_buffer, &byte_offset));
 
+  if (argc == 4) {
+    napi_valuetype valuetype2;
+    NAPI_CALL(env, napi_typeof(env, args[2], &valuetype2));
+
+    NAPI_ASSERT(env, valuetype2 == napi_number,
+        "Wrong type of arguments. Expects a number as third argument.");
+
+    NAPI_CALL(env, napi_get_value_uint32(env, args[2], (uint32_t*)(&length)));
```

Comment:
The cast won't work on big-endian. So something like this maybe?
```C
uint32_t locallength = length;
NAPI_CALL(..., &locallength);
length = locallength;
```

Reply:
Thank you for input. Done.